src/server/middleware/log/log.go

```diff
@@ -0,0 +1,36 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package log
+
+import (
+	"net/http"
+
+	"github.com/goharbor/harbor/src/common/utils/log"
+	"github.com/goharbor/harbor/src/server/middleware"
+)
+
+// Middleware middleware which add logger to context
+func Middleware() func(http.Handler) http.Handler {
+	return middleware.New(func(w http.ResponseWriter, r *http.Request, next http.Handler) {
+		rid := r.Header.Get("X-Request-ID")
+		if rid != "" {
+			ctx := log.WithLogger(r.Context(), log.L.WithFields(log.Fields{"requestID": rid}))
```

Comment:
Would be helpful for debugging if we can log a message with request id and basic information of the request?

Reply:
debug info added